### PR TITLE
Add changable bundle id and docs for macOS signing

### PR DESCRIPTION
**Pull request by ann0see**
**Short description of changes**

Adds variable for a changable Bundle ID for notarization for macOS. One can add the bundle id here: https://github.com/jamulussoftware/jamulus/settings/variables/actions/new

Also adds some more (basic) documentation which needs improvement as soon as signing works, and some clean up.

CHANGELOG: Autobuild: Use variable for Bundle ID on macOS. This allows to change the bundle ID for macOS notarization on GitHub

**Context: Fixes an issue?**

No. Related to: https://github.com/jamulussoftware/jamulus/issues/3255#issuecomment-2295387824

**Does this change need documentation? What needs to be documented and how?**

Maybe the signing - once it works.

**Status of this Pull Request**

Ready for review

**What is missing until this pull request can be merged?**
Review (language wise)

## Checklist

-  [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/main/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
-  [ ] I tested my code and it does what I want (should be merged if it doesn't break. I will try signing on my own repo)
-  [ ] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
-  [ ] I've filled all the content above

### CODE REVIEW

File: .github/workflows/autobuild.yml [hunk 1]
```diff
@@ -365,13 +365,13 @@ jobs:
         run:                        ${{ matrix.config.base_command }} build
         env:
           JAMULUS_BUILD_VERSION:    ${{ needs.create_release.outputs.build_version }}
-          MACOS_CERTIFICATE:        ${{ secrets.MACOS_CERT}}
-          MACOS_CERTIFICATE_PWD:    ${{ secrets.MACOS_CERT_PWD }}
-          MACOS_CERTIFICATE_ID:     ${{ secrets.MACOS_CERT_ID }}
-          MAC_STORE_APP_CERT:       ${{ secrets.MACAPP_CERT}}
+          MACOS_CERTIFICATE:        ${{ secrets.MACOS_CERT }} # Base64 encoded (Developer ID Application (?)) certificate. See https://help.apple.com/xcode/mac/current/#/dev154b28f09
```
Comment (ann0see) [inline .github/workflows/autobuild.yml:368]: Notarization hangs, so not sure if this is the correct cert

**Comment by pljones** [comment]
Will this be in for 3.11.0 release?  If so we should put it on the tracking board as in progress for the release.  As it affects the release process, I feel we should try to target a release...

**Comment by ann0see** [comment]
Yes. Agree

File: .github/workflows/autobuild.yml (hunk 1 above)
Comment (ann0see) [reply .github/workflows/autobuild.yml:368]: Ok. After contacting apple support now it says accepted. The actual bundle ID is still outstanding.

**Comment by ann0see** [comment]
Maybe this will be replaced by another PR soon. I'm refactoring the build script for macOS now.

**Review by softins — APPROVED** [review]
Changes look sensible to me. So happy to approve, even if it might get superseded.

File: .github/workflows/autobuild.yml [hunk 2]
```diff
@@ -365,13 +365,13 @@ jobs:
...
+          MACOS_CERTIFICATE_PWD:    ${{ secrets.MACOS_CERT_PWD }} # Password protecting MACOS_CERTIFICATE
+          MACOS_CERTIFICATE_ID:     ${{ secrets.MACOS_CERT_ID }} # Certificate ID of MACOS_CERTIFICATE. If unknown, import MACOS_CERT into keychain and check the ID there
```
Comment (pljones) [inline .github/workflows/autobuild.yml:370]: Should this be more explicit?
```suggestion
          MACOS_CERTIFICATE_ID:     ${{ secrets.MACOS_CERT_ID }} # Certificate ID of MACOS_CERTIFICATE. If unknown, import secrets.MACOS_CERT into keychain and check the ID there
```
or have I misunderstood the reference?

Comment (ann0see) [reply .github/workflows/autobuild.yml:370]: No. This is OK

**Comment by ann0see** [comment]
Ok. Superseeded by: #3352

**Pull request closed without merging**